src/components/EnterpriseFormComponent.js

```diff
@@ -0,0 +1,265 @@
+'use strict';
+
+import React from 'react';
+import Modal from 'react-modal';
+import { translate } from 'react-i18next';
+
+Modal.setAppElement('#app');
+
+require('styles/EnterpriseForm.scss');
+
+class EnterpriseFormComponent extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      enterprise: props.enterprise,
+      locale: props.locale,
+      modalIsOpen: false
+    }
+
+    this.closeModal = this.closeModal.bind(this);
+    this.deleteEnterprise = this.deleteEnterprise.bind(this);
+    this.handleStringInputChange = this.handleStringInputChange.bind(this);
+    this.handleNumberInputChange = this.handleNumberInputChange.bind(this);
+    this.handleSubmitForm = this.handleSubmitForm.bind(this);
+    this.handleDeleteEnterprise = this.handleDeleteEnterprise.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.enterprise.name !== this.props.enterprise.name) {
+      this.setState({
+        enterprise: nextProps.enterprise
+      });
+    }
+  }
+
+  handleSubmitForm(event) {
+    event.preventDefault();
+
+    let enterprise = this.state.enterprise;
+    const locale = this.state.locale;
+    const endpoint = this.context.config.api_root + '/enterprise/' + enterprise.id;
+    let enterprise_final = {};
```

Comment:
It's not a big deal and hardly worth fixing.. but I noticed a bunch of these so thought I'd mention - isn't JS best practice for style camel case not underscore?